diff -Naur nova-2012.2.2.orig//nova/db/sqlalchemy/models.py nova-2012.2.2//nova/db/sqlalchemy/models.py
--- nova-2012.2.2.orig//nova/db/sqlalchemy/models.py	2013-02-28 13:23:00.000000000 +0100
+++ nova-2012.2.2//nova/db/sqlalchemy/models.py	2013-02-28 13:23:34.000000000 +0100
@@ -882,29 +882,16 @@
     name = Column(String(255))
     availability_zone = Column(String(255), nullable=False)
     _hosts = relationship(AggregateHost,
-                          lazy="joined",
-                          secondary="aggregate_hosts",
                           primaryjoin='and_('
                                  'Aggregate.id == AggregateHost.aggregate_id,'
                                  'AggregateHost.deleted == False,'
-                                 'Aggregate.deleted == False)',
-                         secondaryjoin='and_('
-                                'AggregateHost.aggregate_id == Aggregate.id, '
-                                'AggregateHost.deleted == False,'
-                                'Aggregate.deleted == False)',
-                         backref='aggregates')
+                                 'Aggregate.deleted == False)')
 
     _metadata = relationship(AggregateMetadata,
-                         secondary="aggregate_metadata",
                          primaryjoin='and_('
                              'Aggregate.id == AggregateMetadata.aggregate_id,'
                              'AggregateMetadata.deleted == False,'
-                             'Aggregate.deleted == False)',
-                         secondaryjoin='and_('
-                             'AggregateMetadata.aggregate_id == Aggregate.id, '
-                             'AggregateMetadata.deleted == False,'
-                             'Aggregate.deleted == False)',
-                         backref='aggregates')
+                             'Aggregate.deleted == False)')
 
     @property
     def hosts(self):
