diff -ur nova-2012.2.orig/nova/cloudpipe/pipelib.py nova-2012.2/nova/cloudpipe/pipelib.py
--- nova-2012.2.orig/nova/cloudpipe/pipelib.py	2012-09-26 02:48:35.000000000 +0200
+++ nova-2012.2/nova/cloudpipe/pipelib.py	2012-11-19 15:05:49.000000000 +0100
@@ -52,6 +52,7 @@
                help=_('Netmask to push into openvpn config')),
     ]
 
+flags.DECLARE('cnt_vpn_clients', 'nova.network.manager')
 FLAGS = flags.FLAGS
 FLAGS.register_opts(cloudpipe_opts)
 
