--- setup.py.orig	2012-02-02 12:26:50.000000000 +0100
+++ setup.py	2012-02-02 12:30:53.000000000 +0100
@@ -67,14 +67,12 @@
 ]
 
 config_path = 'etc/quantum/'
-init_path = 'etc/init.d'
 ovs_plugin_config_path = 'etc/quantum/plugins/openvswitch'
 cisco_plugin_config_path = 'etc/quantum/plugins/cisco'
 
 DataFiles = [
     (config_path,
         ['etc/quantum.conf', 'etc/quantum.conf.test', 'etc/plugins.ini']),
-    (init_path, ['etc/init.d/quantum-server']),
     (ovs_plugin_config_path,
         ['etc/quantum/plugins/openvswitch/ovs_quantum_plugin.ini']),
     (cisco_plugin_config_path,
